### PR TITLE
ASTRewrite support for JEP 430

### PR DESCRIPTION
**Pull request by subyssurendran666**
## What it does
https://github.com/eclipse-jdt/eclipse.jdt.core/issues/1800

## Author checklist

- [ ] I have thoroughly tested my changes
- [ ] The change is following the [coding conventions](https://wiki.eclipse.org/Platform/How_to_Contribute#Coding_Conventions)
- [ ] I have signed the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php)

### CODE REVIEW

**Comment by subyssurendran666** [comment]
WIP. Do not merge

**Review by mpalat — COMMENTED** [review]
@subyssurendran666 Let there be test scenarios for all the properties and atleast a test for each of the operations. An Early comment

File: org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/rewrite/describing/ASTRewritingStringTemplateTest.java [hunk 1]
```diff
@@ -35,7 +35,7 @@
 public class ASTRewritingStringTemplateTest extends ASTRewritingTest {
 
 	static {
-//		TESTS_NAMES = new String[] {"test005"};
+		TESTS_NAMES = new String[] {"test007_c"};
```
[the remark this answers is not in the document]
Comment (subyssurendran666) [reply org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/rewrite/describing/ASTRewritingStringTemplateTest.java]: Done

File: org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/rewrite/describing/ASTRewritingStringTemplateTest.java [hunk 2]
```diff
@@ -442,7 +562,123 @@ public void _test007() throws Exception {
...
+		buf.append("    String s = STR.\"\"\"\nHello!\n\"\"\";\n");
+		buf.append("  }\n");
+		buf.append("}\n");
+
+		assertEqualString(preview, buf.toString());
+	}
+
+	@SuppressWarnings({ "rawtypes", "deprecation" })
+	//SINGLE LINE to MULTI LINE with Multiple Components
+	public void test006_c() throws Exception {
```
Comment (mpalat) [inline org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/rewrite/describing/ASTRewritingStringTemplateTest.java:603]: @subyssurendran666 can you please another test with STRing template processor modified with a name having different length than the original template processor

**Comment by mpalat** [comment]
@subyssurendran666 could you please make this patch wrt BETA_JAVA22

**Comment by subyssurendran666** [comment]
> @subyssurendran666 could you please make this patch wrt BETA_JAVA22

Sure Manoj

**Pull request closed without merging**